Created Number Range Summarizer

polybotter opened the pull request.

mbusonhleko commented on impacttest/src/numberrangesummarizer/Summarize.java:
```diff
@@ -0,0 +1,48 @@
+	}
+	
+	@Override
+	public Collection<Integer> collect(String input) {
```
As is (meaning if this method is called directly), this will cause an error for parseInt if a non-Integer is passed through. The regEx to remove all non-Integers should really be in this method to avoid this.

mbusonhleko commented on impacttest/src/numberrangesummarizer/Summarize.java:
```diff
@@ -0,0 +1,48 @@
+package numberrangesummarizer;
```
Your implementation is not in Java 8, which was a requirement.

mbusonhleko commented on impacttest/src/numberrangesummarizer/SummarizeTest.java:
```diff
@@ -0,0 +1,40 @@
+
+class SummarizeTest {
+
+	@Test
```
There are more test cases that could have been handled here:
- How do we handle duplicates?
- How do we handle unsorted input?

polybotter commented:
Hi Mbuso, 

Thank you for taking time to look at the my code, I appreciate the feedback, I have no idea why my eclipse was on the wrong version of Java, corrected it. 

I also thought about adding in the duplicate check and sort after you mentioned it and I like the idea of having it in, so I added it. Also, good point on the regex check, didn't even think about that!

mbusonhleko commented:
Awesome to see that you incorporated my feedback. Just want to clear something up. So the requirement for Java 8 was not really about the Java version as it was about the changes introduced by this version of Java. This includes:
- Stream API
- Lambdas 
My expectation was really to see you utilise this new functionality in your code as a way of demonstrating the ability to understand new changes.

Good work all the same though!
